mobilebot/mobilebot.c

```diff
@@ -164,7 +171,7 @@ void read_mb_sensors(){
     mb_state.left_wheel_distance_delta = mb_state.left_encoder_delta * (WHEEL_DIAMETER * PI) / (GEAR_RATIO * ENCODER_RES);
     mb_state.right_wheel_distance_delta = mb_state.right_encoder_delta * (WHEEL_DIAMETER * PI) / (GEAR_RATIO * ENCODER_RES);
     mb_state.distance_delta = (mb_state.left_wheel_distance_delta + mb_state.right_wheel_distance_delta) / 2;
-
+    mb_state.encoder_yaw_delta = (mb_state.right_wheel_distance_delta - mb_state.left_wheel_distance_delta) / WHEEL_BASE;
```

Comment:
@xunfu I was shouldn't this be a fusion between imu and encoders?

Reply:
so we have two variables recording the change of heading, one is called yaw_delta, which records the heading change based on the imu data, while the other one is called encoder_yaw_delta, which records the heading change based on the encoder data. So in the file mobilebot.c, what is going on is we only update these two variables. The fusion happens inside the function mb_update_odometry in the file mb_odometry.c.

Reply:
@xunfu okay great thank you.